### PR TITLE
[ENG-3676] Remove spammed content from Google indexing

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Popular pages on OSF have tended to be spam, even pages that have already been closed/confirmed spam.  They remain indexed on Google, thus discoverable. For content confirmed spam, remove Google indexing.  For spammed content that is then confirmed ham, return Google indexing.

## Changes

- when a node is not pubic we will add a tag to urge google to remove it from their index.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket

https://openscience.atlassian.net/browse/ENG-3676

### CODE REVIEW

File: website/templates/base.mako [hunk 1]
```diff
@@ -5,11 +5,14 @@
 <head>
     <meta charset="utf-8">
     <title>OSF | ${self.title()}</title>
-    % if settings.GOOGLE_SITE_VERIFICATION:
+    % if settings.GOOGLE_SITE_VERIFICATION and node and node['is_public']:
```
Comment (aaxelb) [inline website/templates/base.mako:8]: are there any non-node, mako-based pages that we still want indexed? can't think of any that haven't been replaced (like osf.io/rr), but that'd be the only worry

**Review by aaxelb — APPROVED** [review]
lgtm -- small concern for unintended consequences, but probably fine 👍

File: website/templates/base.mako (hunk 1 above)
Comment (Johnetordoff) [reply website/templates/base.mako:8]: A quick search yields None, but even if they did most of the metadata uses the `node` data to poulate so on those pages, little of substance is being indexed anyway.

Comment (mfraezz) [reply website/templates/base.mako:8]: There is a `${self.public()}` template function in this file that ostensibly serves the same purpose. I didn't see any use in any template outside of the GA block in this file. Less risk of breaking other things if `node` isn't passed as a templating arg, but more risk of it not working as expected.

Comment (aaxelb) [reply website/templates/base.mako:8]: looks like `${self.public()}` returns a string `in {'public', 'private', 'n/a'}` which we pass to third-party analytics, so probably don't want to reuse for this

could be nice to add `${self.should_amplify_by_search_engine()}` or something that defaults `False`, then move this logic to [`project_base.mako`](https://github.com/CenterForOpenScience/osf.io/blob/HEAD/website/templates/project/project_base.mako), if we want to invest any more attention into mako-template inheritance

**Comment by aaxelb** [comment]
looking at this again... adding the `noindex` tag won't do anything until robot sees it -- how about including recently-deleted content in [the sitemap](https://github.com/CenterForOpenScience/osf.io/blob/develop/scripts/generate_sitemap.py) to prompt a prompt re-crawl?

**Comment by Johnetordoff** [comment]
@aaxelb Yeah definitely agree with you after discussing this more. I'll look into the sitemap.

**Pull request closed without merging**